src/main/scala/com/imaginea/activegrid/core/services/UserService.scala

```diff
@@ -0,0 +1,158 @@
+        node =>
+          val a = userGroup.fromNeo4jGraph(node.getId)
+          println("Mapping = " + a)
+          a
+      }.flatten
+
+    Page[UserGroup](0, listOfUserGroups.size, listOfUserGroups.size, listOfUserGroups)
+  }
+
+  /*
+  * Id parameter is used to fetch the node
+  * Returning Option[UserGroup]
+  */
+  def getUserGroup(id: Long): Future[Option[UserGroup]] = Future {
+    import com.imaginea.activegrid.core.models.UserGroup.RichUserGroup
+
+    val userGroup: UserGroup = null;
+    userGroup.fromNeo4jGraph(id)
+  }
+
+  def deleteUserGroup(userGroupId: Long): Future[Unit] =
+    Future {
+      Neo4jRepository.removeEntity[UserGroupProxy](userGroupId)
+    }
+
+  def getUser(userId: Long): Future[Option[User]] = Future {
+    user.fromNeo4jGraph(userId)
+  }
+
+  def getKeys(userId: Long): Future[Page[KeyPairInfo]] = Future {
+    val keysList = user.fromNeo4jGraph(userId).map(_.publicKeys).getOrElse(List.empty)
+    Page(0, keysList.size, keysList.size, keysList)
+  }
+
+
+  def getKeys(userName: String): Future[Option[Page[KeyPairInfo]]] = Future {
+    logger.debug(s"Searching Users with name ${userName}")
+    val maybeNode = Neo4jRepository.getSingleNodeByLabelAndProperty(label, "username", userName)
+
+    logger.debug(s" May be node ${maybeNode}")
+
+    maybeNode match {
+      case None => {
+        Some(Page(0, 0, 0, List()))
+      }
+      case Some(node) => {
+        val keysList = user.fromNeo4jGraph(node.getId).map(_.publicKeys).getOrElse(List.empty)
+        Some(Page(0, keysList.size, keysList.size, keysList))
+      }
+    }
+
+  }
+
+  def getKey(userId: Long, keyId: Long): Future[Option[KeyPairInfo]] = Future {
+    getKeyById(userId, keyId)
+  }
+
+  def deleteKey(userId: Long, keyId: Long): Future[Option[String]] = Future {
+    logger.debug(s"Deleting Key[${keyId}] of User[${userId}] ")
+    val node = getKeyById(userId, keyId)
+    logger.debug(s"node to be deleted -- ${node}")
+    Some(s"Deleted key with id ${keyId}")
+  }
+
+  def getKeyById(userId: Long, keyId: Long): Option[KeyPairInfo] = {
+    val keysList: List[KeyPairInfo] = user.fromNeo4jGraph(userId).map(_.publicKeys).getOrElse(List.empty)
+
+    keysList match {
+      case keyInfo :: _ if keyInfo.id.get.equals(keyId) => Some(keyInfo)
+      case _ :: keyInfo :: _ if keyInfo.id.get.equals(keyId) => Some(keyInfo)
+      case _ => None
+    }
+  }
+
+  def addKeyPair(userId: Long, sSHKeyContentInfo: SSHKeyContentInfo): Future[Option[Page[KeyPairInfo]]] = Future {
+
+    val keyMaterial = sSHKeyContentInfo.keyMaterials
+
+    keyMaterial.foreach { case (key: String, value: String) => {
+      logger.debug(s" (${key}  --> (${value}))")
+    }
+    }
+
+    //TODO: need to add code to write data to file
+
+    val keysList = user.fromNeo4jGraph(userId).map(_.publicKeys).getOrElse(List.empty)
+    Some(Page(0, keysList.size, keysList.size, keysList))
+  }
+
+  def deleteUser(userId: Long): Future[Option[String]] = Future {
```

Comment:
Seems to be a Merge conflict; The latest Babji's code has not got merged with this.

My changes are available in the flowing userGroup function.
```
 def deleteUserGroup(userGroupId: Long): Future[Unit] =
    Future {
      Neo4jRepository.removeEntity[UserGroupProxy](userGroupId)
    }
```